packages/location-management/src/components/LocationUnitList/index.tsx

```diff
@@ -144,7 +134,7 @@ export const LocationUnitList: React.FC<Props> = (props: Props) => {
       );
       setTableData(data);
     }
-  }, [treeData, currentClickedNode]);
+  }, [treeDataQuery, currentClickedNode]);
```

Comment:
We are watching treeDataQuery here for this effect to run, I think we can move the treeData evaluation in here before the if so that it's only re-evaluated when treeDataQuery changes